pubpub/pubpub
# feature: Allow custom inline content for citations

Resolves #813 by adding a new field to the citation controls to add custom text. Right now it looks a little cramped in there, and I'm going to tweak the styling and solicit some feedback before shipping, but it's functionally all there.

_Test plan:_
- Create a new citation and verify that the usual behavior still works — automatically creating citations from structured data, adding unstructured data, etc.
- Now make sure that you can add and remove custom labels with the new "Inline Content" field.
- Make sure that this information is retained across page refreshes.
- Try this with a non-default inline citation style and ensure that the correct style is rendered into the "Default" state of the inline content controls.

_Screenshot:_
<img width="1278" alt="Screen Shot 2020-05-14 at 7 26 26 PM" src="https://user-images.githubusercontent.com/2208769/81995620-d54a0180-9618-11ea-97e1-9dcf2d4df25d.png">
